internal/service/service.go

```diff
@@ -0,0 +1,146 @@
+package service
+
+import (
+	"errors"
+	"github.com/VTerenya/employees/internal"
+	"github.com/VTerenya/employees/internal/repository"
+)
+
+type Handler interface {
+	CreatePosition(p internal.Position) error
+	CreateEmployee(e internal.Employee) error
+	GetPositions() []internal.Position
+	GetEmployees() []internal.Employee
+	GetPosition(id string) (internal.Position, error)
+	GetEmployee(id string) (internal.Employee, error)
+	DeletePosition(id string) error
+	DeleteEmployee(id string) error
+	UpdatePosition(p internal.Position) error
+	UpdateEmployee(e internal.Employee) error
+}
+
+type Service struct {
+	Handler
+}
+
+func NewService(repository *repository.Repository) *Service {
+	return &Service{
+		newTodoService(repository),
+	}
+}
+
+type todoService struct {
+	repo *repository.Repository
+}
+
+func (t todoService) CreatePosition(p internal.Position) error{
+	m := t.repo.GetPositions()
+	for _, value := range m {
+		if value == p {
+			return errors.New("create error: this position is exists")
+		}
+	}
+	t.repo.AddPosition(p)
+	return nil
+}
+
+func (t todoService) CreateEmployee(e internal.Employee) error{
+	m := t.repo.GetEmployees()
+	for _, value := range m {
+		if value == e {
+			return errors.New("create error: no this employee")
```

Comment:
you have to return the error that employee already exists

Reply:
+